include/deal.II/fe/fe_p1nc.h

```diff
@@ -0,0 +1,311 @@
+ * Note that the set of these values satisfies the dice rule which is described above.
+ * We denote such a function associated with vertex $v_j$ by $\phi_j$.
+ *
+ * The following figures represent $\phi_j$, $j=0,\cdots,3$ with its values at midpoints.
+ * Canonical (local) basis functions are given by any three shape functions among
+ * the following four linear functions.
+ *
+ * <ul>
+ * <li> shape function $\phi_0$:
+ *  @verbatim
+ *  +--------0.0--------+
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ * 0.5                 0.0
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  +--------0.5--------+
+ *  @endverbatim
+ *
+ * <li> shape function $\phi_1$:
+ *  @verbatim
+ *  +--------0.0--------+
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ * 0.0                 0.5
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  +--------0.5--------+
+ *  @endverbatim
+ *
+ * <li> shape function $\phi_2$:
+ *  @verbatim
+ *  +--------0.5--------+
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ * 0.5                 0.0
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  +--------0.0--------+
+ *  @endverbatim
+ *
+ * <li> shape function $\phi_3$:
+ *  @verbatim
+ *  +--------0.5--------+
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ * 0.0                 0.5
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  +--------0.0--------+
+ *  @endverbatim
+ *
+ * </ul>
+ *
+ * Note that above shape functions are constructed on each cell, not on the reference cell only.
+ * @p get_linear_shape computes the coefficients for shape functions when @p fill_fe_values
+ * is called on each cell.
+ *
+ * The basis function associated with a node in global scope is defined by the composition of
+ * (local) basis functions associated with the node on each element.
+ * And DOF associated with the node represents the coefficient of the basis function associated with the node.
+ * In this frame, all DOFs are truly independent.
+ * When a problem with homogeneous Dirichlet boundary condition is considered, the total number of DOFs
+ * is equal to the number of interior nodes, as the number of DOFs which the standard bilinear
+ * finite element @p Q_1 has.
+ *
+ * <h3>Unit support points</h3>
+ * Contrast with ordinary Lagrange finite elements, DOF value with respect to the P1 nonconforming
+ * element at given node does not coincide with the function value at that node.
+ * For instance, the basis function associated with a node has value 0.75 at that node, not 1.0.
```

Comment:
No. it is true for all general quads because of a linear shape function is defined by the midpoint values. 
When a shape function associated with a vertex is given on a quad of any shape, it has value 0.25 on a whole diagonal connecting two adjacent vertices. It gives value 0.75 at the corresponding vertex.